source/initial_conditions/harmonic_perturbation.cc

```diff
@@ -35,12 +35,15 @@ namespace aspect
     HarmonicPerturbation<dim>::
     initial_temperature (const Point<dim> &position) const
     {
+      AssertThrow (this->has_boundary_temperature() || (!use_conductive_profile),
```

Comment:
This might also be a case for using the approach @jaustermann described in #1038 (defining a simple plugin that is called 'conductive profile' and then using that one as a 'base model' for the harmonic perturbation), but lets worry about that later.